src/auth/index.js

```diff
@@ -350,6 +350,145 @@ export default class Auth {
+      {
+        parameters: {
+          mfaToken: {required: true, toName: 'mfa_token'},
+          oobCode: {required: true, toName: 'oob_code'},
+          bindingCode: {required: false, toName: 'binding_code'},
+        },
+      },
+      parameters,
+    );
+
+    return this.client
+      .post('/oauth/token', {
+        ...payload,
+        client_id: this.clientId,
+        grant_type: 'http://auth0.com/oauth/grant-type/mfa-oob',
+      })
+      .then(responseHandler);
+  }
+
+  /**
+   * Log in a user using a multi-factor authentication Recovery Code after they have received the 'mfa_required' error.
+   * The MFA token tells the server the username or email, password, and realm values sent on the first request.
+   *
+   * Requires your client to have the **MFA** Grant Type enabled. See [Client Grant Types](https://auth0.com/docs/clients/client-grant-types) to learn how to enable it.
+   *
+   * @param {Object} parameters login with Recovery Code parameters
+   * @param {String} parameters.mfaToken the token received in the previous login response
+   * @param {String} parameters.recoveryCode the recovery code provided by the end-user.
+   * @returns {Promise}
+   *
+   * @memberof Auth
+   */
+  loginWithRecoveryCode(parameters = {}) {
+    const payload = apply(
+      {
+        parameters: {
+          mfaToken: {required: true, toName: 'mfa_token'},
+          recoveryCode: {required: true, toName: 'recovery_code'},
+        },
+      },
+      parameters,
+    );
+
+    return this.client
+      .post('/oauth/token', {
+        ...payload,
+        client_id: this.clientId,
+        grant_type: 'http://auth0.com/oauth/grant-type/mfa-recovery-code',
+      })
+      .then(responseHandler);
+  }
+
+  /**
+   * Log in a user using the One Time Password code after they have received the 'mfa_required' error.
+   * The MFA token tells the server the username or email, password, and realm values sent on the first request.
+   *
+   * Requires your client to have the **MFA OTP** Grant Type enabled.
+   * See [Client Grant Types](https://auth0.com/docs/clients/client-grant-types) to learn how to enable it.
+   *
+   * @param {Object} parameters challenge request parameters
+   * @param {String} parameters.mfaToken the token received in the previous login response
+   * @param {String} parameters.challengeType A whitespace-separated list of the challenges types accepted by your application.
+   * Accepted challenge types are oob or otp. Excluding this parameter means that your client application
+   * accepts all supported challenge types.
+   * @param {String} parameters.authenticatorId The ID of the authenticator to challenge.
+   * @returns {Promise}
+   *
+   * @memberof Auth
+   */
+  multifactorChallenge(parameters = {}) {
+    const payload = apply(
+      {
+        parameters: {
+          mfaToken: {required: true, toName: 'mfa_token'},
+          challengeType: {required: true, toName: 'challenge_type'},
```

Comment:
`challengeType` and `authenticatorId` are not required: https://auth0.com/docs/api/authentication#challenge-request